sdk/js/src/utils/repairVaa.ts

```diff
@@ -0,0 +1,137 @@
+import { ethers } from "ethers";
+import { CONTRACTS } from "./consts";
+import { Implementation__factory } from "../ethers-contracts";
+import { parseVaa, GuardianSignature } from "../vaa";
+import { hexToUint8Array } from "./array";
+import { keccak256 } from "../utils";
+
+const ETHEREUM_CORE_BRIDGE = CONTRACTS["MAINNET"].ethereum.core;
+
+function hex(x: string): string {
+  return ethers.utils.hexlify(x, { allowMissingPrefix: true });
+}
+
+export async function getGuardianSets(
+  provider: ethers.providers.JsonRpcProvider
+) {
+  let result: any = {};
+  const core = Implementation__factory.connect(ETHEREUM_CORE_BRIDGE, provider);
+  (result.address = ETHEREUM_CORE_BRIDGE),
+    (result.currentGuardianSetIndex = await core.getCurrentGuardianSetIndex());
+  result.guardianSet = {};
+  for (let i of Array(result.currentGuardianSetIndex + 1).keys()) {
+    let guardian_set = await core.getGuardianSet(i);
+    result.guardianSet[i] = { keys: guardian_set[0], expiry: guardian_set[1] };
+  }
+  return result;
+}
+
+export async function repairVaa(vaaHex: string, rpcUrl: string) {
```

Comment:
would be good to have a pure version of this function that doesn't hit the rpc and instead just takes in the guardian set as an argument. Then another version can wrap the pure one and do the actual RPC query. Simpler to understand in separation